conformance/lang/expressions/member-access-expr/array_member_access_expr.balt

```diff
@@ -0,0 +1,594 @@
+Test-Case: parser-error
+Description: Test missing key expression in member access expression.
+Labels: array-type, int, list-constructor-expr, member-access-expr
+
+function errorFunction() {
+    int[] arr = [1, 2, 3];
+    _ = arr[]; // @error missing key expression
+}
+
+Test-Case: output
+Description: Test whether the static type of key expression is subtype of int when the basic type of the
+             container expression is an array.
+Labels: array-type, int, list-constructor-expr, member-access-expr, module-const-decl, module-type-defn, union-type
+
+const i1 = 0;
+const i2 = 5;
+
+type IntType1 3|4;
+
+type IntType2 i1|i2;
+
+function init() {
+    int[] arr = [10, 20, 30, 40, 50, 60, 70, 80];
+
+    int i0 = 0;
+    io:println(arr[i0]); // @output 10
+
+    IntType1 i3 = 4;
+    io:println(arr[i3]); // @output 50
+
+    io:println(arr[i1]); // @output 10
+
+    IntType2 i4 = 5;
+    io:println(arr[i4]); // @output 60
+
+    io:println(arr[5]); // @output 60
+}
+
+Test-Case: output
+Description: Test whether the static type of key expression is subtype of int when the basic type of the
+             container expression is an array.
+Fail-Issue: ballerina-platform/ballerina-lang#33346
```

Comment:
https://github.com/ballerina-platform/ballerina-lang/issues/33346 is fixed on 2201.0.x